### PR TITLE
Refactoring codes

### PR DESCRIPTION
**Pull request by osamingo**
The title said all.

### CODE REVIEW

**Comment by osamingo** [comment]
@dsnet Please review it if you are free.

**Comment by osamingo** [comment]
@dsnet Please review it if you are free.

File: cmp/report_slices.go [hunk 1]
```diff
@@ -322,7 +322,7 @@ func coalesceInterveningIdentical(groups []diffStats, windowSize int) []diffStat
 			hadX, hadY := prev.NumRemoved > 0, prev.NumInserted > 0
 			hasX, hasY := next.NumRemoved > 0, next.NumInserted > 0
 			if ((hadX || hasX) && (hadY || hasY)) && curr.NumIdentical <= windowSize {
-				*prev = (*prev).Append(*curr).Append(*next)
+				*prev = prev.Append(*curr).Append(*next)
```
Comment (dsnet) [inline cmp/report_slices.go]: Already covered by #149.

File: cmp/internal/diff/diff.go [hunk 2]
```diff
@@ -93,9 +93,8 @@ type Result struct{ NumSame, NumDiff int }
 func BoolResult(b bool) Result {
 	if b {
 		return Result{NumSame: 1} // Equal, Similar
-	} else {
-		return Result{NumDiff: 2} // Not Equal, not Similar
 	}
+	return Result{NumDiff: 2} // Not Equal, not Similar
```
Comment (dsnet) [inline cmp/internal/diff/diff.go]: This is deliberate.

Both branches have equal probability of being executed. They are kept at the same indentation for that purpose.

File: cmp/internal/value/zero.go [hunk 3]
```diff
@@ -11,7 +11,7 @@ import "reflect"
 func IsZero(v reflect.Value) bool {
 	switch v.Kind() {
 	case reflect.Bool:
-		return v.Bool() == false
+		return !v.Bool()
```
Comment (dsnet) [inline cmp/internal/value/zero.go]: This is deliberate. It keeps consistency with the code below.

File: cmp/report_compare.go [hunk 4]
```diff
@@ -89,13 +89,13 @@ func (opts formatOptions) FormatDiff(v *valueNode) textNode {
 			if v.NumDiff == 0 {
 				outx := opts.FormatValue(v.ValueX, visitedPointers{})
 				outy := opts.FormatValue(v.ValueY, visitedPointers{})
-				if v.NumIgnored > 0 && v.NumSame == 0 {
+				switch {
+				case v.NumIgnored > 0 && v.NumSame == 0:
 					return textEllipsis
-				} else if outx.Len() < outy.Len() {
+				case outx.Len() < outy.Len():
```
Comment (dsnet) [inline cmp/report_compare.go]: This is deliberate.

Both branches have equal probability of being executed. Keeping them on the same indent highlights this fact.

File: cmp/report_compare.go [hunk 5]
```diff
@@ -167,9 +166,7 @@ func (opts formatOptions) formatDiffList(recs []reportRecord, k reflect.Kind) te
 			if k == reflect.Struct {
 				var isZero bool
 				switch opts.DiffMode {
-				case diffIdentical:
-					isZero = value.IsZero(r.Value.ValueX) || value.IsZero(r.Value.ValueX)
-				case diffRemoved:
+				case diffIdentical, diffRemoved:
```
Comment (dsnet) [inline cmp/report_compare.go]: This is actually a bug, fixed in #148.

File: cmp/report_text.go [hunk 6]
```diff
@@ -365,7 +365,7 @@ func (s diffStats) String() string {
 	// Pluralize the name (adjusting for some obscure English grammar rules).
 	name := s.Name
 	if sum > 1 {
-		name = name + "s"
+		name += "s"
```
Comment (dsnet) [inline cmp/report_text.go]: Fixed in #149.

File: cmp/cmpopts/struct_filter.go [hunk 7]
```diff
@@ -170,7 +170,7 @@ func canonicalName(t reflect.Type, sel string) ([]string, error) {
 	if !ok {
 		return []string{name}, fmt.Errorf("does not exist")
 	}
-	var ss []string
+	ss := make([]string,0,len(sf.Index))
```
Comment (dsnet) [inline cmp/cmpopts/struct_filter.go]: I'm not fond of up-front slice allocation unless the code has shown a performance benefit from doing so.
None of these slice usages are in the hot-path. Thus, the readability loss is not worth the supposed performance gains.

**Comment by dsnet** [comment]
Thanks for the PR. Some of the changes made:
* are un-doing deliberate style choices made by the original author
* are already been fixed by another contributor

After those two, there isn't much left to this PR.

**Comment by osamingo** [comment]
@dsnet 
No problem, thanks for your review. I feel nice other PRs were merged :)
Please publish the new release, for Go module.

**Pull request closed without merging**